### PR TITLE
chore: clone jerni store mongodb to the repo

### PR DESCRIPTION
**Pull request by votrungquan1999**
## Epic:
- Solve the issue of memory overflow on the Partners Portal

## Planned Tasks:
- Clone MongoDB Store into the mono repo  ***(You are here 👈)***
- Persist received events into a database
- Handle edge cases:
  -  When there are new events in the subscription list
  - Resume subscription at the last saved event id

## In this PR
- [x] Clone MongoDB Store into the mono-repo
- [x] Change tests from Ava to Bun
- [x] Add prettier rules

### CODE REVIEW

**Comment by votrungquan1999** [comment]
* **#1** <a href="https://app.graphite.dev/github/pr/tungv/jerni-3/1?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2** <a href="https://app.graphite.dev/github/pr/tungv/jerni-3/2?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @votrungquan1999 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/jerni-store-mongodb/tests/optimistic_locking/update.spec.ts [hunk 1]
```diff
@@ -1,354 +1,332 @@
...
-  t.deepEqual(
-    result.map((x) => x.name),
-    ["test_4", "------", "------", "test_1"],
-  );
-  t.deepEqual(
-    result.map((x) => x.counter),
-    [0, 1, 1, 0],
-  );
-});
+  test.only("it should not apply an updateOne twice with `pipeline`", async () => {
```
Comment (votrungquan1999) [inline packages/jerni-store-mongodb/tests/optimistic_locking/update.spec.ts:86]: remove .only

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (votrungquan1999) [inline .prettierrc.json:1]: change to biome

File: packages/jerni/tsconfig.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "moduleDetection": "force",
     "allowImportingTsExtensions": true,
     "strict": true,
-    "strictNullCheck": true,
+    "strictNullChecks": true,
     "downlevelIteration": true,
```
Comment (votrungquan1999) [inline packages/jerni/tsconfig.json:11]: aadd nullable index strict

**Review by tungv — APPROVED** [review]
